projects/storefrontlib/src/lib/cms/components/dynamic-slot/dynamic-slot.component.ts

```diff
@@ -88,9 +89,14 @@ export class DynamicSlotComponent implements OnInit {
    *
    * CAUTION: This function should not be used for SmartEdit bindings.
    */
-  getComponentType(component: ContentSlotComponentData): string {
-    return component.typeCode === JSP_INCLUDE_CMS_COMPONENT_TYPE
-      ? component.uid
-      : component.typeCode;
+  getComponentMappedType(component: ContentSlotComponentData): string {
+    switch (component.typeCode) {
+      case JSP_INCLUDE_CMS_COMPONENT_TYPE:
+        return component.uid;
+      case FLEX_CMS_COMPONENT_TYPE:
```

Comment:
component inside slot only has "uid", "uuid" and "componentType". "flextype" is not saved in "slot/component". It is part of component data. So, we have to put "flextype" into store slot/component.